src/java.security.jgss/windows/classes/sun/security/krb5/internal/tools/Klist.java

```diff
@@ -78,115 +78,113 @@ public class Klist {
+                if (args[i].equals("-?") ||
+                        args[i].equals("-h") ||
+                        args[i].equals("--help")) {
+                    printHelp();
+                    return 0;
+                }
+                if ((args[i].length() >= 2) && (args[i].startsWith("-"))) {
+                    arg = Character.valueOf(args[i].charAt(1));
+                    switch (arg.charValue()) {
+                        case 'c':
+                            action = 'c';
+                            break;
+                        case 'k':
+                            action = 'k';
+                            break;
+                        case 'a':
+                            options[2] = 'a';
+                            break;
+                        case 'n':
+                            options[3] = 'n';
+                            break;
+                        case 'f':
+                            options[1] = 'f';
+                            break;
+                        case 'e':
+                            options[0] = 'e';
+                            break;
+                        case 'K':
+                            options[1] = 'K';
+                            break;
+                        case 't':
+                            options[2] = 't';
+                            break;
+                        default:
+                            System.out.println("Invalid argument: " + args[i]);
+                            printHelp();
+                            return -1;
+                    }
+                } else {
+                    if (!args[i].startsWith("-") && (i == args.length - 1)) {
+                        // the argument is the last one.
+                        name = args[i];
+                    } else {
+                        System.out.println("Invalid argument: " + args[i]);
+                        printHelp(); // incorrect input format.
+                        return -1;
+                    }
+                }
+            }
         }
-        switch (klist.action) {
+        switch (action) {
         case 'c':
-            if (klist.name == null) {
-                klist.target = CredentialsCache.getInstance();
-                klist.name = CredentialsCache.cacheName();
+            if (name == null) {
+                target = CredentialsCache.getInstance();
+                name = CredentialsCache.cacheName();
             } else
-                klist.target = CredentialsCache.getInstance(klist.name);
+                target = CredentialsCache.getInstance(name);
 
-            if (klist.target != null)  {
-                klist.displayCache();
+            if (target != null)  {
+                return displayCache();
             } else {
-                klist.displayMessage("Credentials cache");
-                System.exit(-1);
+                displayMessage("Credentials cache");
+                return -1;
```

Comment:
Make displayMessage(String) return -1 and just use "return displayMessage();"? (Other displayXXX() also returns int)